src/Bridge/NelmioApiDoc/Extractor/AnnotationsProvider/ApiPlatformProvider.php

```diff
@@ -0,0 +1,239 @@
+            $data['output'] = sprintf('%s:%s', ApiPlatformParser::OUT_PREFIX, $resourceClass);
+        }
+
+        if ($collection && Request::METHOD_GET === $method) {
+            $resourceFilters = $resourceMetadata->getCollectionOperationAttribute($operationName, 'filters', [], true);
+
+            $data['filters'] = [];
+            foreach ($this->filters as $filterName => $filter) {
+                if (in_array($filterName, $resourceFilters)) {
+                    foreach ($filter->getDescription($resource) as $name => $definition) {
+                        $data['filters'][] = ['name' => $name] + $definition;
+                    }
+                }
+            }
+        }
+
+        $apiDoc = new ApiDoc($data);
+        $apiDoc->setRoute($route);
+
+        return $apiDoc;
+    }
+
+    /**
+     * Gets Hydra documentation for the given resource.
+     *
+     * @param array  $hydraApiDoc
+     * @param string $prefixedShortName
+     *
+     * @return array|null
+     */
+    private function getResourceHydraDoc(array $hydraApiDoc, string $prefixedShortName)
+    {
+        foreach ($hydraApiDoc['hydra:supportedClass'] as $supportedClass) {
+            if ($supportedClass['@id'] === $prefixedShortName) {
+                return $supportedClass;
+            }
+        }
+    }
+
+    /**
+     * Gets the Hydra documentation of a given operation.
+     *
+     * @param string $method
+     * @param array  $hydraDoc
+     *
+     * @return array|null
+     */
+    private function getOperationHydraDoc(string $method, array $hydraDoc)
+    {
+        foreach ($hydraDoc['hydra:supportedOperation'] as $supportedOperation) {
+            if ($supportedOperation['hydra:method'] === $method) {
+                return $supportedOperation;
+            }
+        }
+    }
+
+    /**
+     * Gets the Hydra documentation for the collection operation.
+     *
+     * @param string $shortName
+     * @param string $method
+     * @param array  $hydraEntrypointDoc
+     *
+     * @return array|null
+     */
+    private function getCollectionOperationHydraDoc(string $shortName, string $method, array $hydraEntrypointDoc)
+    {
+        $propertyName = '#Entrypoint/'.lcfirst($shortName);
+
+        foreach ($hydraEntrypointDoc['hydra:supportedProperty'] as $supportedProperty) {
+            $hydraProperty = $supportedProperty['hydra:property'];
+            if ($hydraProperty['@id'] === $propertyName) {
+                return $this->getOperationHydraDoc($method, $hydraProperty);
+            }
+        }
+    }
+
+    /**
+     * Finds the route for an operation on a resource.
+     *
+     * @param string $resourceClass
+     * @param bool   $collection
+     * @param string $operationName
+     *
+     * @return Route
+     *
+     * @throws \InvalidArgumentException
+     */
+    private function getRoute(string $resourceClass, bool $collection, string $operationName) : Route
```

Comment:
What do you think about creating a new `public function getOperationRoute()` in https://github.com/api-platform/core/blob/master/src/Bridge/Symfony/Routing/OperationMethodResolver.php#L55-L95 (without adding it to the interface). Then type hint the constructor with `OperationMethodResolver` from the Symfony bridge in the constructor (Nelmio already depends of the Symfony router so it's not a problem IMO).

It will avoid a code duplication and add a new useful method that can be used in other contexts in the Symfony Method resolver.